guardian/prosemirror-elements
# style: Ergonomic type renaming

## What does this change?

This PR aims to improve the readability of the library by making a few helpful renames to important types:

Field -> FieldDescription
FieldSpecs -> FieldDescriptions

FieldViewSpec -> Field
 Properties: 
 - fieldSpec -> description
 - fieldView -> view

Field (React component) -> FieldWrapper

This should make setting up new elements and their views easier to understand intuitively and reasoning with the type system more natural. 

## How to test

This change should be a no-op and make no functional changes. Consider how this renamed types and arguments affect how easy it is to understand the api and create new elements. 

## How can we measure success?

Comparing our API and type system to our [vision document](https://github.com/guardian/prosemirror-elements/blob/main/docs/vision.md), the plugin should  "abstract the complexity of adding nested content and provide developers with a clear intuitive interface".
